TestingHelpers/MockFile.cs

```diff
@@ -525,12 +525,30 @@ public override IEnumerable<string> ReadLines(string path, Encoding encoding)
 #if NET40
         public override void Replace(string sourceFileName, string destinationFileName, string destinationBackupFileName)
         {
-            throw new NotImplementedException(StringResources.Manager.GetString("NOT_IMPLEMENTED_EXCEPTION"));
+            Replace(sourceFileName, destinationFileName, destinationBackupFileName, false);
         }
 
         public override void Replace(string sourceFileName, string destinationFileName, string destinationBackupFileName, bool ignoreMetadataErrors)
         {
-            throw new NotImplementedException(StringResources.Manager.GetString("NOT_IMPLEMENTED_EXCEPTION"));
+            if (!mockFileDataAccessor.FileExists(sourceFileName))
```

Comment:
Looking at the documentation for Replace(), a ArgumentNullException is thrown when the destFileName parameter is null.

Reply:
Added explicit null checks